### PR TITLE
Set subscriptions archived when user is deactivated;

### PR DESCRIPTION
**Pull request by marceloschmidt**
@RocketChat/core 

Closes #ISSUE_NUMBER

### CODE REVIEW

File: packages/rocketchat-lib/client/lib/roomTypes.coffee [hunk 1]
```diff
@@ -42,6 +42,13 @@ RocketChat.roomTypes = new class roomTypesClient extends roomTypesCommon
 
 		return room?.ro is true and Array.isArray(room?.muted) and room?.muted.indexOf(user.username) != -1
 
+	archived: (roomId, user) ->
```
Comment (rodrigok) [inline packages/rocketchat-lib/client/lib/roomTypes.coffee]: Are you using this param `user`?

File: packages/rocketchat-lib/server/models/Subscriptions.coffee [hunk 2]
```diff
@@ -307,6 +307,17 @@ RocketChat.models.Subscriptions = new class extends RocketChat.models._Base
...
+	setArchivedByUsername: (username, archived) ->
+		query =
+			t: 'd'
+			name: username
+
+		update =
+			$set:
+				archived: archived
+
+		return @update query, update
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/models/Subscriptions.coffee]: Are you missing `multi: true` here?

File: server/startup/migrations/v061.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+RocketChat.Migrations.add({
+	version: 61,
+	up: function() {
+		RocketChat.models.Users.find({ active: false }).forEach(function(user) {
+			RocketChat.models.Subscriptions.setArchivedByUsername(user.username, false);
```
Comment (rodrigok) [inline server/startup/migrations/v061.js]: Shouldn't be `true` on setArchivedByUsername?

File: packages/rocketchat-lib/client/lib/roomTypes.coffee (hunk 1 above)
Comment (marceloschmidt) [inline packages/rocketchat-lib/client/lib/roomTypes.coffee]: No, I'll remove it.

File: packages/rocketchat-lib/server/models/Subscriptions.coffee (hunk 2 above)
Comment (marceloschmidt) [inline packages/rocketchat-lib/server/models/Subscriptions.coffee]: yes, nice catch again :)

File: server/startup/migrations/v061.js (hunk 3 above)
Comment (marceloschmidt) [inline server/startup/migrations/v061.js]: And yes again :D

**Comment by marceloschmidt** [comment]
@rodrigok once again proving that peer-review is an important part of development ;D thanks!

**Comment by rodrigok** [comment]
LGTM

**Comment by geekgonecrazy** [comment]
lgtm 👍

**Pull request merged**